### DISCUSSION TITLE
Hypervisor with OpenAMP

### DISCUSSION THREAD

**Discussion by pedromvas** [OpenAMP/open-amp#476] [General]
Hi,

I am new to this embedded linux scenario but i wanted to build a system in a xilinx ZCU106 that can handle some form of mixed criticality.  In this case I was wondering if it would be possible to use some kind of hypervisor on the A53 cpu and to run so commands on the R5 cpu at the same time.

**Comment by arnopo** [comment]
Hello,
@TanmayShah-xilinx any clues?

**Comment by tnmysh** [comment]
Hello,

You could run Xen on a53. From Xen  you could start/stop R5.
What kind of commands do you want to run on R5?

Thanks.

**Reply by pedromvas** [reply]
I thought of maybe reading a sensor and writing the values on a file that can be accessed by a DomU VM

**Reply by tnmysh** [reply]
Yes It is possible. You can use proxy service on RPU side and create file on linux.

Please take a look at proxy_app explanation here: https://openamp.readthedocs.io/en/latest/openamp-system-reference/examples/linux/rpmsg-proxy-app/README.html

Demo RPU firmware code is here: https://github.com/OpenAMP/open-amp/blob/main/apps/examples/rpc_demo/rpc_demo.c
